### PR TITLE
ENG-3044: Introduce support for set streaming

### PR DESCRIPTION
**Pull request by erickpintor**
Changes: 

* New event type;
* New event handler callback;
* New valid field;
* Test cases.

### CODE REVIEW

File: faunadb/client.py [hunk 1]
```diff
@@ -314,6 +314,7 @@ def stream(self, expression, options=None, on_start=None, on_error=None, on_vers
         subscription.on('error', on_error)
         subscription.on('version', on_version)
         subscription.on('history_rewrite', on_history)
+        subscription.on('set', on_set)
```
Comment (faunaee) [inline faunadb/client.py:317]: The doc string above needs to be updated to include an entry for `on_set`.

**Review by faunaee — CHANGES_REQUESTED** [review]
Seems to work well. Just need to update a doc string accordingly.

File: faunadb/streams/client.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 from .errors import StreamError
 from .events import Error, parse_stream_request_result_or_none
 
-VALID_FIELDS = {"diff", "prev", "document", "action"}
+VALID_FIELDS = {"diff", "prev", "document", "action", "set"}
```
Comment (faunaee) [inline faunadb/streams/client.py:17]: Isn't the added field `index`? I don't see a test for the stream option fields. When I try to use `"index"`, I get:

```
Exception: Valid fields options are {'action', 'prev', 'document', 'diff', 'set'}, provided ['action', 'document', 'index', 'prev', 'diff'].
```

**Review by faunaee — CHANGES_REQUESTED** [review]
Possible type in field definition.

File: faunadb/streams/client.py (hunk 2 above)
Comment (erickpintor) [reply faunadb/streams/client.py:17]: You're right. That was a typo. Thank you.

**Review by faunaee — DISMISSED** [review]
LGTM! 👍

**Review by elersong — APPROVED** [review]

**Pull request merged**